include/RAJA/DepGraphNode.hxx

```diff
@@ -119,6 +108,33 @@ public:
   int& semaphoreReloadValue() { return m_semaphore_reload_value; }
 
   ///
+  /// Ready this task to be used again
+  ///
+  void reset() { m_semaphore_value.store(m_semaphore_reload_value); }
+
+  ///
+  /// Satisfy one incoming dependency
+  ///
+  void satisfyOne()
+  {
+    if (m_semaphore_value > 0) {
+      int val = --m_semaphore_value;
+    }
+  }
+
+  ///
+  /// Wait for all dependencies to be satisfied
+  ///
+  void wait()
+  {
+    while (m_semaphore_value > 0) {
+      // TODO: an efficient wait would be better here, but the standard
+      // promise/future is not good enough
+      std::this_thread::yield();
```

Comment:
It certainly wouldn't hurt, but this is really a stopgap at the moment.  At least for OpenMP, this should probably tie into the built-in task mechanism rather than being built on top, the same may be true for the other models as well.